src/components/New Calendar/Calendar.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+import { DayNameRow } from './DayNameRow';
+import { DateTile } from './DateTile';
+import { Text, Layout } from '@ui-kitten/components';
+import { compareDates } from '../../services/Calendar/helper_functions';
+import { dateTileDimensions } from '../../shared/constants';
+
+interface CalendarProps {
+    displayedMonth: Date;
+    onTilePress?: (date) => void;
+    selectedDates?;
+    initialTasks;
+}
+
+export const Calendar = ({
+    displayedMonth,
+    onTilePress,
+    selectedDates = [],
+    initialTasks,
+}: CalendarProps) => {
+    const month = displayedMonth.getMonth();
+    const year = displayedMonth.getFullYear();
+    const initialDate = new Date(year, month, 1);
+    const dateArray = new Array(6);
+    const today = new Date();
+
+    function determineRenderDate(initial): () => Date {
+        //sets start of week to correct day
+        let renderDate: Date = new Date(
+            initial.setDate(initial.getDate() - initial.getDay()),
+        );
+
+        return updateDate;
+
+        function updateDate() {
+            const returnDate: Date = new Date(renderDate);
+            renderDate.setDate(renderDate.getDate() + 1);
+            return returnDate;
+        }
+    }
+
+    const determineDate: () => Date = determineRenderDate(initialDate);
```

Comment:
this could then be called:
```suggestion
    const getNextDate: () => Date = determineRenderDate(initialDate);
```

Reply:
i like it